prefect_firebolt/database.py

```diff
@@ -95,3 +99,57 @@ async def get_connection(self) -> Connection:
             api_endpoint=self.credentials.api_endpoint,
             additional_parameters=self.additional_parameters,
         )
+
+
+@task
+async def query_firebolt(
+    database: FireboltDatabase,
+    query: str,
+    parameters: "Optional[Sequence[ParameterType]]" = None,
+) -> "List[List[ColType]]":
+    """
+    Executes a query against a Firebolt database.
+
+    Args:
+        database: Firebolt database configuration to use for query execution.
+        query: SQL query to execute.
+        parameters: A sequence of substitution parameters. Used to replace `?`
+            placeholders inside a query with actual values
+
+    Returns:
+        All rows retrieved by the query.
+
+    Example:
+        Execute a query against a Firebolt database:
+        ```python
+        from prefect import flow
+
+        from prefect_firebolt import FireboltCredentials, FireboltDatabase
+        from prefect_firebolt.database import query_firebolt
+
+
+        @flow
+        def run_firebolt_query():
+            firebolt_database_block = FireboltDatabase(
+                database="intergalactic_trips",
+                credentials=FireboltCredentials(
+                    username="arthur.dent@hitchhikers.com", password="dont42panic"
+                ),
+            )
+
+            results = query_firebolt(
+                database=firebolt_database_block,
+                query="SELECT * FROM ex_intergalactic_trips LIMIT 100",
+            )
+
+            return results
+
+
+        run_firebolt_query()
+        ```
+    """
+    async with await database.get_connection() as connection:
+        cursor = connection.cursor()
+        with cursor:
+            await cursor.execute(query=query, parameters=parameters)
```

Comment:
Do we need async_execution:
https://python-sdk.docs.firebolt.io/en/latest/firebolt.db.html#module-firebolt.db.cursor

Reply:
My understanding is that would result in server-side asynchronous execution which isn't necessary since this doesn't block client side. Plus it looks like Firebolt doesn't support getting results from server-side asynchronous executions.